cmd/entrypoint/env.go

```diff
@@ -206,7 +188,7 @@ func GetDiagdFlags(ctx context.Context, demoMode bool) []string {
 	// XXX: this was not in entrypoint.sh
 	result = append(result, "--port", GetDiagdBindPort())
 
-	cdir := GetConfigDir(demoMode)
+	cdir := env("AMBASSADOR_CONFIG_DIR", path.Join(GetAmbassadorConfigBaseDir(), "ambassador-config"))
```

Comment:
Do we need this at all anymore? What still uses the filesystem flag for diagd?

Reply:
I don't know of anything that does, but we still have one endpoint that listens for a request and enables it

Reply:
What do you mean by `endpoint`? 

Looking around the directory for `ambassador-confg` isn't used at all throughout the repository so I think it is essentially legacy code. It apparently is a `Values.ambassadorConfig` in the Helm chart but other than that there is no reference to it. Do you have any context on it? It is probably a separate PR from this one but I'm thinking we should remove it as well because we don't actively support/document it or know if it even works. 

https://github.com/emissary-ingress/emissary/blob/b9657518353b87dc05fef18a4c06b58e252da4dc/charts/emissary-ingress/templates/deployment.yaml#L233

Reply:
As we discussed offline, lets resolve that one independently of this PR.